src/event/EventPropagators.js

```diff
@@ -40,10 +38,9 @@ function listenerAtPhase(id, event, propagationPhase) {
  */
 function accumulateDirectionalDispatches(domID, upwards, event) {
   if (__DEV__) {
-    warning(
-      domID,
-      'Dispatching id must not be null'
-    );
+    if (!domID) {
+      throw new Error('Dispatching id must not be null');
+    }
```

Comment:
Didn't we intentionally change this to not throw to bring dev & prod behavior in line? (#2868)